src/item.cpp

```diff
@@ -2867,6 +2867,27 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
                     }
                 }
             }
+        } else if( is_gun() ) {
+
+            std::string bp_name;
+            bodypart_id bp_id;
+            for( const item *mod : gunmods() ) {
+                if( has_flag( flag_IS_ARMOR ) ) {
```

Comment:
Are you checking the gun item itself, or the gunmod item?

Reply:
oh wow, yeah the for loop should have been inside that if (has_flag already checks for inherited flags from gunmods)

i probably should remove the for loop ... it isn't needed, at least until we get gunmods that require specific flag checks to be associated with a body part other than torso.